### PR TITLE
Improve completion of methods arguments

### PR DESCRIPTION
**Pull request by jalvesaq**
- Look for methods for second and following classes in objects with multiple classes.
- Always run code to build completion data as a Neovim job.
- Build the `args_` file right after building the corresponding `objls_` file.
- Add time to build `objls_` + `args_` to `:RDebugInfo`.

### CODE REVIEW

**Comment by jalvesaq** [comment]
This pull request should be tried with the branch `improve_methods_args` from `cmp-r`.

Building an `args_` file was about three times slower than building the corresponding `objls_` file. After some optimization, it's now about 5 times faster than before. The big improvement in the performance was due to avoiding calling `help()` just to get the directory where the package was installed. Now that it's much faster, it is being built right after the `objls_` file. This simplifies the code, which now has about 260 fewer lines.

**Comment by PMassicotte** [comment]
oh let me try it!

**Comment by PMassicotte** [comment]
image

Thas is nice!

**Comment by PMassicotte** [comment]
Are we missing some parameters?

image

**Comment by jalvesaq** [comment]
`correlation` is for `summary.lm`, not for `print.summary.lm`

**Comment by PMassicotte** [comment]
oh wow, I still need coffee this morning :)

I have to make some change to a project, I will report back in 1 h ( I will use your PR)

**Comment by jalvesaq** [comment]
In this example, the R documentation is shared by two functions and the `Arguments` section combine all arguments of the two functions.

**Comment by PMassicotte** [comment]
This feels very fast already

**Comment by jalvesaq** [comment]
The completion by `rnvimserver` + `cmp-r` is almost instantaneous. The only slowness is making the cache files when the library is loaded for the first time.

File: nvimcom/R/misc.R [hunk 1]
```diff
@@ -427,26 +427,28 @@ nvim.getmethod <- function(fname, objclass) {
...
-                                     } else {
-                                         return(" = ")
-                                     })
-                env <- paste0(fname, "\x02", fnm)
-                clpstr <- paste0( "', cls = 'm', env = '", env, "'}, {label = '")
-                luastr <- paste0(names(luatbl), unname(luatbl),
-                                 collapse = clpstr)
-                luastr <- paste0("{{label = '", luastr, "', cls = 'm', env = '", env, "'}}")
-                return(luastr)
+        for (obc in objclass) {
```
Comment (PMassicotte) [inline nvimcom/R/misc.R:430]: Nicely done!

**Comment by PMassicotte** [comment]
Well, I have worked roughly 1 hour and it works perfectly fine.  Speed is not comparable as it was before. You did a very nice job here :) I do not see any problem at mergin this :+1:

**Comment by jalvesaq** [comment]
I ran it with Valgrind and there was no invalid read or write in the C code. So, I believe that it's safe to merge. If any bugs appear, we will fix them.

**Comment by PMassicotte** [comment]
I have tried it also on a Window machine and no issue.

**Comment by jalvesaq** [comment]
Thank you!

**Comment by PMassicotte** [comment]
Just need to merge the cmp branch

**Comment by jalvesaq** [comment]
I'll do it... Thanks for remembering!

**Pull request merged**